splunk/tests/splunk_test_prep.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// BuildSplunk builds the Splunk image used for testing Splunk queries.
+func BuildSplunk() {
+	buildCmd := exec.Command("podman", "build", "-t", "test_splunk", "./splunk")
+
+	buildCmd.Stdout = os.Stdout
+	buildCmd.Stderr = os.Stderr
+	err := buildCmd.Run()
+	if err != nil {
+		log.Fatalf("Failed to build image: %v", err)
+	}
+	log.Println("Image build done successfully!")
+}
+
+// RunSplunk deploy the Splunk container used for testing Splunk queries.
+func RunSplunk() {
+	filePath := "runtime.txt"
+	file, err := os.Create(filePath)
+	if err != nil {
+		log.Fatalf("Failed to create file: %v", err)
+	}
+	defer file.Close()
+
+	cmd := exec.Command("podman", "run", "-e", "SPLUNK_START_ARGS=--accept-license", "-e", "SPLUNK_PASSWORD=Password", "-p", "8000:8000", "-p", "8089:8089", "localhost/test_splunk:latest")
```

Comment:
We probably can't assume the image is there aleady, perhaps we should use the `kube play` thing which can also build the images as needed.

Reply:
I read a bit, not sure how kube play is coming in hand here? 
also can't we assume it when we build it a moment before? are we trying to avoid a race condition?

Reply:
kube play can automatically build a missing image when an image reference is a local path

> Kube play is capable of building images on the fly given the correct directory layout and Containerfiles. This option is not available for remote clients, including Mac and Windows (excluding WSL2) machines, yet. Consider the following excerpt from a YAML file:

Reply:
Seems like things work a bit different using kube play, there the second stage of the dockerfile must kill the splunkd process from the first stage, otherwise the 8089 port stays occupied

Reply:
Changed to use podman kube play. This demanded killing the splunkd process during the second stage of the build, as mentioned in the former comment. 
only issue left with this approach is that the manifest has to have a name for the pod - meaning we won't be able to use it concurrently.